File: app/admin/service/internal/data/dict_type.go
package data

import (
	"context"
	"time"

	"github.com/go-kratos/kratos/v2/log"
	"github.com/hypercoze/kratos-admin/app/admin/service/internal/biz"
	"github.com/hypercoze/kratos-admin/app/admin/service/internal/data/ent"
	"github.com/hypercoze/kratos-admin/app/admin/service/internal/data/ent/dictdata"
	"github.com/hypercoze/kratos-admin/app/admin/service/internal/data/ent/dicttype"
	"github.com/hypercoze/kratos-admin/pkg/enthelper"
)

var _ biz.DictTypeRepo = (*dictTypeRepo)(nil)

type dictTypeRepo struct {
	data *Data
	log  *log.Helper
}

func NewDictTypeRepo(data *Data, logger log.Logger) biz.DictTypeRepo {
	return &dictTypeRepo{
		data: data,
		log:  log.NewHelper(logger),
	}
}

// 列表（支持分页）
func (r *dictTypeRepo) ListDictType(ctx context.Context, params *biz.ListDictTypeRequest, opts ...*biz.ListDictTypeOption) (*biz.ListDictTypeResponse, error) {
	query := r.data.db.DictType.Query().
		Order(ent.Desc(dicttype.FieldCreatedAt))

	opt := &biz.ListDictTypeOption{}
	if len(opts) > 0 {
		opt = opts[0]
	}

	if opt.OnlyDeleted {
		query = query.Where(dicttype.DeletedAtNotNil())
	} else if !opt.IncludeDeleted {
		query = query.Where(dicttype.DeletedAtIsNil())
	}

	if v := params.Name; len(v) > 0 {
		query = query.Where(dicttype.NameContains(v))
	}
	if v := params.Code; len(v) > 0 {
		query = query.Where(dicttype.CodeContains(v))
	}
	if v := params.Status; len(v) > 0 {
		query = query.Where(dicttype.StatusEQ(dicttype.Status(v)))
	}

	res, err := enthelper.Pagination[*ent.DictType, *ent.DictTypeQuery](ctx, query, params.PaginationParams)
	if err != nil {
		return nil, err
	}

	// 转换为biz结构
	data := make([]*biz.DictType, 0, res.Total)
	// 转换为biz结构
	for _, v := range res.Data {
		data = append(data, &biz.DictType{
			ID:        v.ID,
			Name:      v.Name,
			Code:      v.Code,
			Status:    string(v.Status),
			Remark:    v.Remark,
			CreatedAt: v.CreatedAt,
			UpdatedAt: v.UpdatedAt,
			DeletedAt: v.DeletedAt,
		})
	}

	return &biz.ListDictTypeResponse{
		Data:  data,
		Total: res.Total,
	}, nil
}

// 创建
func (r *dictTypeRepo) CreateDictType(ctx context.Context, d *biz.DictType) error {
	_, err := r.data.db.DictType.Create().
		SetID(d.ID).
		SetName(d.Name).
		SetCode(d.Code).
		SetStatus(dicttype.Status(d.Status)).
		SetNillableRemark(&d.Remark).
		SetCreatedAt(d.CreatedAt).
		SetUpdatedAt(d.UpdatedAt).
		Save(ctx)
	return err
}

// 编辑
func (r *dictTypeRepo) UpdateDictType(ctx context.Context, d *biz.DictType) error {
	_, err := r.data.db.DictType.UpdateOneID(d.ID).
		SetName(d.Name).
		SetCode(d.Code).
		SetStatus(dicttype.Status(d.Status)).
		SetNillableRemark(&d.Remark).
		SetUpdatedAt(d.UpdatedAt).
		Save(ctx)
	return err
}

// 删除
func (r *dictTypeRepo) DeleteDictType(ctx context.Context, ids []string) error {
	if len(ids) == 0 {
		return nil
	}

	now := time.Now()
	if err := r.data.db.DictType.
		Update().
		Where(dicttype.IDIn(ids...)).
		SetDeletedAt(now).
		Exec(ctx); err != nil {
		return err
	}

	return r.data.db.DictData.
		Update().
		Where(dictdata.DictTypeIDIn(ids...)).
		SetDeletedAt(now).
		Exec(ctx)
}

func (r *dictTypeRepo) UpdateDictTypeStatus(ctx context.Context, id, status string) error {
	_, err := r.data.db.DictType.UpdateOneID(id).
		SetStatus(dicttype.Status(status)).
		SetUpdatedAt(time.Now()).
		Save(ctx)
	return err
}

File: frontend/apps/web-antd/src/api/system/dictionary/dict-type.ts
import { requestClient } from '#/api/request';
import type { PaginationParams,PaginationResult } from '#/types/pagination'

export namespace DictionaryDictTypeApi {
  /** 字典类型 */
  export interface DictType {
    /** 类型ID */
    id: string;
    /** 类型名称 */
    name: string;
    /** 类型编码 */
    code: string;
    /** 状态：enabled=启用 disabled=禁用 */
    status: 'enabled' | 'disabled';
    /** 备注 */
    remark?: string;
    /** 创建时间 */
    createdAt?: string;
    /** 更新时间 */
    updatedAt?: string;
  }

  export interface DictTypeListParams extends PaginationParams{

  }
}

/**
 * 获取字典类型列表
 */
async function getDictTypeListApi(params?:DictionaryDictTypeApi.DictTypeListParams) {
  return requestClient.get<PaginationResult<DictionaryDictTypeApi.DictType>>('/admin/v1/dict-type',{
    params:params
  });
}

/**
 * 获取字典类型
 */
async function getDictTypeApi(id:string) {
  return requestClient.get<DictionaryDictTypeApi.DictType>(
    `/admin/v1/dict-type/${id}`,
  );
}

/**
 * 创建字典类型
 */
async function createDictTypeApi(
  data: Omit<DictionaryDictTypeApi.DictType, 'id' | 'createdAt' | 'updatedAt'>,
) {
  return requestClient.post('/admin/v1/dict-type', data,{
    showSuccessMessage: true,
  });
}

/**
 * 更新字典类型
 */
async function updateDictTypeApi(
  id: string,
  data: Omit<DictionaryDictTypeApi.DictType, 'id' | 'createdAt' | 'updatedAt'>,
) {
  return requestClient.put(`/admin/v1/dict-type/${id}`, data,{
    showSuccessMessage: true,
  });
}

/**
 * 更新字典类型状态
 */
async function updateDictTypeStatusApi(
  id: string,
  status: DictionaryDictTypeApi.DictType['status'],
) {
  return requestClient.put(`/admin/v1/dict-type/${id}/status`, {
    status,
  });
}

/**
 * 删除字典类型
 */
async function deleteDictTypeApi(ids: string[]) {
  return requestClient.delete('/admin/v1/dict-type', {
    params: {
      ids,
    },
  });
}

export {
  getDictTypeListApi,
  getDictTypeApi,
  createDictTypeApi,
  updateDictTypeApi,
  updateDictTypeStatusApi,
  deleteDictTypeApi,
};

